contrib/platform/src/com/sun/jna/platform/win32/COM/Accessible.java

```diff
@@ -0,0 +1,77 @@
+/* Copyright (c) 2021 Mo Beigi, All Rights Reserved
+ *
+ * The contents of this file is dual-licensed under 2
+ * alternative Open Source/Free licenses: LGPL 2.1 or later and
+ * Apache License 2.0. (starting with JNA version 4.0.0).
+ *
+ * You can freely decide which license you want to apply to
+ * the project.
+ *
+ * You may obtain a copy of the LGPL License at:
+ *
+ * http://www.gnu.org/licenses/licenses.html
+ *
+ * A copy is also included in the downloadable source code package
+ * containing JNA, in file "LGPL2.1".
+ *
+ * You may obtain a copy of the Apache License at:
+ *
+ * http://www.apache.org/licenses/
+ *
+ * A copy is also included in the downloadable source code package
+ * containing JNA, in file "AL2.0".
+ */
+package com.sun.jna.platform.win32.COM;
+
+import com.sun.jna.Pointer;
+import com.sun.jna.platform.win32.Variant.VARIANT;
+import com.sun.jna.platform.win32.WTypes.BSTRByReference;
+import com.sun.jna.platform.win32.WinNT.HRESULT;
+import com.sun.jna.ptr.IntByReference;
+
+/**
+ * Implementation for {@link IAccessible } interface
+ *
+ * @author Mo Beigi, me@mobeigi.com
+ */
+public class Accessible extends Dispatch implements IAccessible
+{
+
+    public Accessible(Pointer pvInstance) { super(pvInstance); }
+
+    public HRESULT get_accName(VARIANT varChild, BSTRByReference pszName)
+    {
+        return (HRESULT) this._invokeNativeObject(10,
+                new Object[] { this.getPointer(), varChild, pszName }, HRESULT.class);
+    }
+
+    public HRESULT get_accValue(VARIANT varChild, BSTRByReference pszName)
+    {
+        return (HRESULT) this._invokeNativeObject(11,
+                new Object[] { this.getPointer(), varChild, pszName }, HRESULT.class);
+    }
+
+    public HRESULT get_accRole(VARIANT varChild, VARIANT.ByReference pvarRole)
+    {
+        return (HRESULT) this._invokeNativeObject(13,
+                new Object[] { this.getPointer(), varChild, pvarRole }, HRESULT.class);
+    }
+
+    public HRESULT get_accChildCount(IntByReference pcountChildren)
+    {
+        return (HRESULT) this._invokeNativeObject(8,
+                new Object[] { this.getPointer(), pcountChildren }, HRESULT.class);
+    }
+
+    public HRESULT get_accDefaultAction(VARIANT varChild, BSTRByReference pszDefaultAction)
+    {
+        return (HRESULT) this._invokeNativeObject(20,
+                new Object[] { this.getPointer(), varChild, pszDefaultAction }, HRESULT.class);
+    }
+
+    public HRESULT accDoDefaultAction(VARIANT varChild)
+    {
+        return (HRESULT) this._invokeNativeObject(25,
```

Comment:
Wouldn't it be much nicer if all the `VTable` offsets were defined in an enum or just as `private static ints` at the top of this file? Would be a nice assist to future contributors too who may want to expand the mapping and add an extra function.

Reply:
A list of the vtbl functions in a constant is probably sufficient to document the order.

Personally I prefer the format in the [Converter](https://github.com/java-native-access/jna/blob/7dbbf4b447674182e5752ac9b1eba91d9bffc22e/contrib/platform/src/com/sun/jna/platform/win32/COM/IShellFolder.java#L454) class in `IShellFolder`.   Declare an array of pointers and read the COM function pointers into it, then access the function pointers as `vTable[i]`.  

You could probably get fancy with an enum aligned with the array, e.g., `vTable[ACC_DO_DEFAULT_ACTION.ordinal()]` but I think that overly complicates it.

Reply:
Glanced around the internet and saw another interesting approach which actually probably aligns closest with the native implementation:  similar to the pointer array in my previous comment, declare an `AccessibleVtbl` extending the `Structure` class with the appropriately named function pointers as the fields.   

You'd instantiate an instance of the structure, e.g., `AccessibleVtbl vtbl = new AccessibleVtbl(ptr);` and then instead of `vtable[i]` in the example linked above the function pointer would be `vtbl.accDoDefaultAction`.